### ISSUE TITLE
Configure: version number is missing from ENGINESDIR in Makefile

### ISSUE THREAD

**Issue by mspncp** [openssl/openssl#8071]
After configuring, the version number is missing from `ENGINESDIR` variable in the Makefile

```Makefile
ENGINESDIR=$(libdir)/engines-
```

As a consequence, OpenSSL is unable to find it's installed engines:

```
msp@msppc:~/src/openssl$ strace -f openssl rand -engine foo |& grep engine
execve("/opt/openssl-dev/bin/openssl", ["openssl", "rand", "-engine", "foo"], 0x7fff36c86fc0 /* 52 vars */) = 0
openat(AT_FDCWD, "/opt/openssl-dev/lib/engines-/foo.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
write(2, "invalid engine \"foo\"\n", 21invalid engine "foo"
...
```

### Analysis

The version number is missing because the `$sover_dirname` is not determined correctly by the configure script. A bisection shows that this bug was introduced in commit 3a63dbef15b62b121c5df8762f8cb915fb06b27a, which introduced the new versioning scheme.

For the fun of it and as an exercise, I wrote a little bisection script (my first one :-) ) which configures OpenSSL and checks the `ENGINESDIR` variable.

**engine-checker.sh**

```sh
#!/bin/bash

(make clean && rm Makefile && perl Configure debug-linux-x86_64) >/dev/null || (echo CONFIGURE FAILED ; exit 125)

if grep '^ENGINESDIR=.*/engines-[0-9.]\+' Makefile; then
	echo GOOD; exit 0
else
	echo BAD; exit 1
fi
```

If you want to try it out yourself, save it to your openssl source directory and make it executable:

```
~/src/openssl$  chmod +x engine-checker.sh
```

Now you can run a fully automated bisection:

```
~/src/openssl$ git bisect start master OpenSSL_1_1_1
Bisecting: 260 revisions left to test after this (roughly 8 steps)
[4b801fdcf4c25f44374eb18cb18f36d904975edd] Add an error message test recipes for system error messages

~/src/openssl$ git bisect run ./engine-checker.sh
running ./engine-checker.sh

ENGINESDIR=$(libdir)/engines-1.1
GOOD
Bisecting: 130 revisions left to test after this (roughly 7 steps)
[42ea4ef2db123b4060d6d1b92556416c5a6ce2a1] CHANGES: Add Linux Kernel TLS data-path
running ./engine-checker.sh
BAD
Bisecting: 64 revisions left to test after this (roughly 6 steps)
[5477e84273e5877fad53d6b3fd794ed80c2c9524] Following the license change, modify the boilerplates in crypto/ct/
running ./engine-checker.sh
BAD
Bisecting: 32 revisions left to test after this (roughly 5 steps)
[3da2e9c4ee45989a426ff513dc6c6250d1e460de] bn/bn_div.c: make conditional addition unconditional
running ./engine-checker.sh
ENGINESDIR=$(libdir)/engines-1.1
GOOD
Bisecting: 16 revisions left to test after this (roughly 4 steps)
[2c18d164f52a77b9ecc1f835b83aec476334ebbc] Following the license change, modify the boilerplates in ssl/
running ./engine-checker.sh
BAD
Bisecting: 7 revisions left to test after this (roughly 3 steps)
[478d4f912c001f18717d1364132184ff9d31f944] include/openssl/opensslconf.h.in: allow future deprecation for v4.0
running ./engine-checker.sh
BAD
Bisecting: 3 revisions left to test after this (roughly 2 steps)
[2284f64c871c2f7611678dba6eadd1989cee9981] util/mkdef.pl: prepare for DEPRECATEDIN_X
running ./engine-checker.sh
ENGINESDIR=$(libdir)/engines-1.1
GOOD
Bisecting: 1 revision left to test after this (roughly 1 step)
[3a63dbef15b62b121c5df8762f8cb915fb06b27a] Switch to MAJOR.MINOR.PATCH versioning and version 3.0.0-dev
running ./engine-checker.sh
BAD
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[672f943ad6c6d16b1f65a77b8e2c83c8f44a112b] Switch future deprecation version from 1.2.0 to 3.0
running ./engine-checker.sh
ENGINESDIR=$(libdir)/engines-1.1
GOOD
3a63dbef15b62b121c5df8762f8cb915fb06b27a is the first bad commit
commit 3a63dbef15b62b121c5df8762f8cb915fb06b27a
Author: Richard Levitte <levitte@openssl.org>
Date:   Thu Sep 27 15:56:35 2018 +0200

    Switch to MAJOR.MINOR.PATCH versioning and version 3.0.0-dev
    
    We're strictly use version numbers of the form MAJOR.MINOR.PATCH.
    Letter releases are things of days past.
    
    The most central change is that we now express the version number with
    three macros, one for each part of the version number:
    
        OPENSSL_VERSION_MAJOR
        OPENSSL_VERSION_MINOR
        OPENSSL_VERSION_PATCH
    
    <... snip ...>

    (Merged from https://github.com/openssl/openssl/pull/7724)

:100644 100644 c55457509b8b552f611a9cdfbaf66c6ebcabd034 e31b087b4a0f2eff46f4876a2c6a48ad98616ebe M	CHANGES
:040000 040000 4cc7fc5bad3b14d82682259ec23cf1c2b2dbb75c 180632d87c5049f49acbb50bbe46e71ec3a37625 M	Configurations
:100755 100755 65bbec1f312cd4c50eba909dd04b005862487d9c f9515a945c36647aac11b39ba4639979f1bf25fc M	Configure
:100644 100644 df16b7894f9a79a9ebf295675e580e7d7b320bb2 7ac249e5f591ec1d5d7f64a36783d975f5521a07 M	NEWS
:100644 100644 6818807a4cbd1aaa454c7adf005bbc9519ea05a7 fe3f98070c2de4611f397c7e6311bf203b55d8c6 M	README
:040000 040000 e0508055dc924b413f8e748909f2cbe5ba99fb56 5bee35a859b56cb70e7c8a6f05c285477a2da8b3 M	VMS
:040000 040000 1c877635552ec52d060a0367d3420b0cc9ae01fb 7904608f6a2a6330dc57360749993c27665aeae1 M	apps
:100644 100644 c2ed667afb04af36f24f82922085b41f5de24cbd afcb772caffb3457fddef4dd3061d9f4aa079313 M	build.info
:040000 040000 f677bd67ed6b7a38628be4b6840a55663cadd0c2 16a2ddf8943e9ab6589e1fb41e6bebca0fe3d448 M	crypto
:040000 040000 9e5107223bc3c9242c35258b96c51c3b70f733a9 139e1951b003ad7831701899c06f39576da83738 M	doc
:040000 040000 da00f7751936f70832a119b4aa8c5329fe7f55ed f78afd5ccf4614a29e93b18fc7d55154481efde1 M	engines
:040000 040000 faff69be603a6b552e42860a0b519ef57062bd0d d5b911580739d78efc398d106583f352d51fed83 M	include
:040000 040000 f8ef16b632c513e29c1a4490cbb01263b4bb4df8 63b88735829115911d692cc01fdc0ab0e12dc5d4 M	test
:040000 040000 1b4a45da936b8b3f867d37cb3cab9e4abd2d6dd3 862822c56e40a023bb75a6c264b099367f8468ad M	util
bisect run success
~/src/openssl$ 
```

**Comment by mspncp** [comment]
I just ran the complete testsuite with full tracing: It is still failing but I didn't check the details yet.

_Side note: I had t run the tests in a tmux session, because it's output was so noisy that I regularly got kicked out of my ssh session. ;-)_

**Comment by levitte** [comment]
I believe that was fixed in 61db9961417e74cbd4a285fe482f1f2b30c5536b

**Comment by levitte** [comment]
A lot of configuration development happened between 3a63dbe and 61db996, so the issue that was fixed with the latter wasn't even related with this issue.  And yet, the symptoms were the same.  So I guess this issue got fixed by accident :wink:
